rd-net/Lifetimes/Lifetimes/LifetimeDefinition.cs

```diff
@@ -89,7 +91,32 @@ private int ThreadLocalExecuting(int increment = 0)
     }
     
     private const int WaitForExecutingInTerminationTimeoutMsDefault = 500;
-    [PublicAPI] public static int WaitForExecutingInTerminationTimeoutMs = WaitForExecutingInTerminationTimeoutMsDefault;  
+    [PublicAPI] public static int WaitForExecutingInTerminationTimeoutMs = WaitForExecutingInTerminationTimeoutMsDefault;
+
+    /// <summary>
+    /// Gets the actual value in milliseconds for semantic (short, long, etc) termination timeout.
+    /// </summary>
+    /// <param name="timeout">semantic timeout as defined by <see cref="LifetimeTerminationTimeout"/></param>
+    /// <returns>timeout value in milliseconds</returns>
+    [PublicAPI]
+    public static int GetTerminationTimeoutMs(LifetimeTerminationTimeout timeout) =>
+      timeout == LifetimeTerminationTimeout.Default
+        ? WaitForExecutingInTerminationTimeoutMs
+        : ourTerminationTimeoutMs[(int)timeout - 1];
+
+    /// <summary>
+    /// Sets the actual value in milliseconds for semantic (short, long, etc) termination timeout.
+    /// </summary>
+    /// <param name="timeout">semantic timeout as defined by <see cref="LifetimeTerminationTimeout"/></param>
+    /// <param name="milliseconds">timeout value in milliseconds</param>
+    [PublicAPI]
+    public static void SetTerminationTimeoutMs(LifetimeTerminationTimeout timeout, int milliseconds)
```

Comment:
Why so complicated? Why not to have 4 -values array with timeouts and always slice it?

Reply:
I'm not sure... Do you mean why `timeout == Default ? WaitForExecutingInTerminationTimeoutMs : ourTerminationTimeoutMs[(int)timeout - 1];` instead of `ourTerminationTimeoutMs[timeout];`?
This is due to backward compatibility - the `WaitForExecutingInTerminationTimeoutMs` is a part of public API.

Reply:
>  ...create a lifetime that will be shutting down slowly and the world will have to wait
It is possible anyway - we can wait any time in any termination action. Our current implementation of AsyncLifetime does exactly this - waits in OnTermination action.

Reply:
> It is possible anyway - we can wait any time in any termination action. Our current implementation of AsyncLifetime does exactly this - waits in OnTermination action.

Lifetime waits for the timeout then issues warning and finishes its Terminate method. In your case it will wait up to 30 seconds.

Reply:
> why can't we use ourTerminationTimeoutMs[timeout];? when ourTerminationTimeoutMs contains 4 values?

Because the `WaitForExecutingInTerminationTimeoutMs` is a field and we can't store its actual value in array, and can't remove `WaitForExecutingInTerminationTimeoutMs` or convert to property because it is part of public API.

Reply:
> Lifetime waits for the timeout then issues warning and finishes its Terminate method. In your case it will wait up to 30 seconds.

If some action which was added via OnTermination() method blocks for arbitrary time, the Terminate() method will not finish till action completion. So, anyway it is possible to block termination for any arbitrary time.

Reply:
The change in the library is ok anyway. When we use long timout in the product and it affects product's shutdown we'll deal with it